src/registry/routes/component-info.js

```diff
@@ -4,61 +4,82 @@ var parseAuthor = require('parse-author');
+  var author = component.author || {};
+  return _.isString(author) ? parseAuthor(author) : author;
+}
 
-      if(err){
-        res.errorDetails = err;
-        return res.status(404).json({ err: err });
+function addGetRepositoryUrlFunction(component) {
+  component.getRepositoryUrl = function() {
+    if (_.isObject(this.repository)) {
+      if (this.repository.url) {
+        return this.repository.url;
       }
+    }
+    if (_.isString(this.repository)) {
+      return this.repository;
+    }
+    return null;
+  };
+}
 
-      var isHtmlRequest = !!req.headers.accept && req.headers.accept.indexOf('text/html') >= 0;
+function componentInfo(err, req, res, component) {
+  if(err) {
+    res.errorDetails = err.localError;
+    return res.status(404).json(err);
+  }
 
-      if(isHtmlRequest && !!res.conf.discovery){
+  var isHtmlRequest = !!req.headers.accept && req.headers.accept.indexOf('text/html') >= 0;
 
-        var params = {},
-            author = component.author || {},
-            parsedAuthor = _.isString(author) ? parseAuthor(author) : author;
+  if(isHtmlRequest && !!res.conf.discovery){
 
-        if(!!component.oc.parameters){
-          var mandatoryParams = _.filter(_.keys(component.oc.parameters), function(paramName){
-            var param = component.oc.parameters[paramName];
-            return !!param.mandatory && !!param.example;
-          });
+    var params = getParams(component);
+    var parsedAuthor = getParsedAuthor(component);
+    addGetRepositoryUrlFunction(component);
 
-          params = _.mapObject(_.pick(component.oc.parameters, mandatoryParams), function(param){
-            return param.example;
-          });
-        }
+    return res.render('component-info', {
+      component: component,
+      dependencies: _.keys(component.dependencies),
+      href: res.conf.baseUrl,
+      parsedAuthor: parsedAuthor,
+      sandBoxDefaultQs: urlBuilder.queryString(params)
+    });
 
-        component.getRepositoryUrl = function() {
-          if (_.isObject(this.repository)) {
-            if (this.repository.url) {
-              return this.repository.url;
-            }
-          }
-          if (_.isString(this.repository)) {
-            return this.repository;
-          }
-          return null;
-        };
+  } else {
+    res.status(200).json(_.extend(component, {
+      requestVersion: req.params.componentVersion || ''
+    }));
+  }
+}
 
-        return res.render('component-info', {
-          component: component,
-          dependencies: _.keys(component.dependencies),
-          href: res.conf.baseUrl,
-          parsedAuthor: parsedAuthor,
-          sandBoxDefaultQs: urlBuilder.queryString(params)
+module.exports = function(conf, repository){
+  return function(req, res){
+    repository.getComponent(req.params.componentName, req.params.componentVersion, function(localRegistryError, localComponent){
+      if(localRegistryError && conf.fallbackRegistryUrl) {
```

Comment:
One little thing here in regards of naming. In many other places in the code, we refer "local" registry as registry running as oc dev, as way to differentiate it from the "default" registry running as real instance.

I think what you are calling local here can be confused with the other "local" so I think I would prefer us to change the naming. What about making local => default (so `registryError`, `component`) and the other default => fallback (so `fallbackError`, `fallbackComponent`)?

Another idea is to call the "local" "current"? Or maybe you have other ideas?